File: src/parser/nim/actions/cache.nim
import std/[os, asyncdispatch, json, strformat]

import ../utils/[chalk, config, paths]

proc nlcli_cache*(s: string = "{}") {.async.} =
    let cachepath = paths["cachepath"]

    let jdata = parseJSON(s)
    let clear = jdata{"clear"}.getBool()
    var level = jdata{"level"}.getStr()

    initconfig()

    # Clear cache...
    if dirExists(cachepath) and clear:
        const filetypes = { pcFile, pcLinkToFile }
        for kind, path in walkDir(cachepath):
            if kind in filetypes: removeFile(path)
        let success = "success".chalk("green")
        echo fmt"{success} Cleared cache."

    if level.len != 0:
        if level[0] in {'0' .. '9'}:
            if level[0] notin {'0' .. '2'}: level = "1"
            setsetting("cache", level)
        else:
            stdout.write(getsetting("cache"))

File: src/parser/js2/actions/cache.js
"use strict";

const path = require("path");
const chalk = require("chalk");
const flatry = require("flatry");
const de = require("directory-exists");
const { paths, readdir, remove, hasProp } = require("../utils/toolbox.js");
const { initconfig, getsetting, setsetting } = require("../utils/config.js");

module.exports = async (args) => {
	let { cachepath } = paths;
	let { clear, level } = args;

	await initconfig();

	if (clear) {
		let [err, res] = await flatry(de(cachepath));
		if (!err) {
			let [err, files] = await flatry(readdir(cachepath));
			if (!files) files = [];
			for (let i = 0, l = files.length; i < l; i++) {
				await remove(path.join(cachepath, files[i]));
			}
			console.log(chalk.green("success"), "Cleared cache");
		}
	}

	if (hasProp(args, "level")) {
		if (Number.isInteger(level)) {
			const levels = [0, 1, 2]; // Cache levels.
			await setsetting("cache", -~levels.indexOf(level) ? level : 1);
		} else {
			process.stdout.write(await getsetting("cache"));
		}
	}
};
